pkg/odo/genericclioptions/util.go

```diff
@@ -34,32 +31,40 @@ func getFirstChildOfCommand(command *cobra.Command) *cobra.Command {
 
 // checkProjectCreateOrDeleteOnlyOnInvalidNamespace errors out if user is trying to create or delete something other than project
 // errFormatForCommand must contain one %s
-func checkProjectCreateOrDeleteOnlyOnInvalidNamespace(command *cobra.Command, errFormatForCommand string) {
+func checkProjectCreateOrDeleteOnlyOnInvalidNamespace(command *cobra.Command, errFormatForCommand string) error {
 	// do not error out when its odo delete -a, so that we let users delete the local config on missing namespace
 	if command.HasParent() && command.Parent().Name() != "project" && (command.Name() == "create" || (command.Name() == "delete" && !command.Flags().Changed("all"))) {
 		err := fmt.Errorf(errFormatForCommand, command.Root().Name())
-		util.LogErrorAndExit(err, "")
+		if err != nil {
+			return err
+		}
```

Comment:
I think here you can simply do `return fmt.Errorf(errFormatForCommand, command.Root().Name())`.

Reply:
But it's the same, right? err is defined in the above line in the same way we can just return err.

Reply:
It is the same, but unnecessary.

Reply:
Understood! I have pushed the required change.